packages/aws-cdk/test/api/hotswap/state-machine-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,478 @@
+import { StepFunctions } from 'aws-sdk';
+import * as setup from './hotswap-test-setup';
+
+let mockUpdateMachineDefinition: (params: StepFunctions.Types.UpdateStateMachineInput) => StepFunctions.Types.UpdateStateMachineOutput;
+
+beforeEach(() => {
+  setup.setupHotswapTests();
+  mockUpdateMachineDefinition = jest.fn();
+  setup.setUpdateStateMachineMock(mockUpdateMachineDefinition);
+});
+
+test('returns undefined when a new StateMachine is added to the Stack', async () => {
+  // GIVEN
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+});
+
+test('calls the updateStateMachine() API when it receives only a definitionString change without Fn::Join in a state machine', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: '{ Prop: "old-value" }',
+          StateMachineName: 'my-machine',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: '{ Prop: "new-value" }',
+            StateMachineName: 'my-machine',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateMachineDefinition).toHaveBeenCalledWith({
+    definition: '{ Prop: "new-value" }',
+    stateMachineArn: 'my-machine',
+  });
+});
+
+test('calls the updateStateMachine() API when it receives only a definitionString change with Fn::Join in a state machine', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value", "AnotherProp" : "another-old-value" }',
+              ],
+            ],
+          },
```

Comment:
Same comment here - use the `Fn::Join` for something here, not just a longer way to write `'{ "Prop" : "old-value", "AnotherProp" : "another-old-value" }'`.